src/analysis/modals/ComputeModal/AzureComputeModal/AzureComputeModal.js

```diff
@@ -83,6 +93,34 @@ export const AzureComputeModalBase = ({
     refreshRuntime();
   }, [currentRuntime, location, onError, workspaceId]);
 
+  // Once all data is loaded, check if default computeConfig need to be adjusted based on billing project limits.
+  useEffect(() => {
+    // If there's an existing runtime, computeConfig will be based on its configuration and should be not be changed.
+    if (!loading && !currentRuntimeDetails && resourceLimits) {
+      if (resourceLimits.availableMachineTypes) {
+        const selectedMachineType = computeConfig.machineType;
+        if (!resourceLimits.availableMachineTypes.includes(selectedMachineType)) {
+          updateComputeConfig('machinetypes', resourceLimits.availableMachineTypes[0]);
+        }
+      }
+
+      if (resourceLimits.maxAutopause) {
+        const autopauseConfig = computeConfig.autopauseThreshold;
+        updateComputeConfig('autopauseThreshold', _.clamp(minAutopause, resourceLimits.maxAutopause, autopauseConfig));
```

Comment:
Can we add some kind of message telling the user that there's a max autopause if it exists?  Otherwise it just reverts to a different number than you typed and is kinda confusing and obnoxious

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4904/commits/6b211c25ec3fa59d9a51f4b43667292db902ceaf.